front50-core/src/main/groovy/com/netflix/spinnaker/front50/model/StorageServiceSupport.java

```diff
@@ -330,22 +330,33 @@ private String buildObjectKey(String id) {
             .from(ids)
             .flatMap(entry -> {
                   try {
-                    T object = (T) service.loadObject(objectType, entry.getKey());
+                    String key = entry.getKey();
+                    T object = (T) service.loadObject(objectType, key);
 
-                    Long expectedLastModifiedTime = keyUpdateTime.get(entry.getKey());
+                    Long expectedLastModifiedTime = keyUpdateTime.get(key);
                     Long currentLastModifiedTime = object.getLastModified();
 
                     if (expectedLastModifiedTime != null && currentLastModifiedTime != null) {
                       if (currentLastModifiedTime < expectedLastModifiedTime) {
                         log.warn(
                           "Unexpected stale read for {} (current: {}, expected: {})",
-                          entry.getKey(),
+                          key,
                           new Date(currentLastModifiedTime),
                           new Date(expectedLastModifiedTime)
                         );
                       }
                     }
 
+                    if (!key.equals(buildObjectKey(object))) {
+                      log.warn(
```

Comment:
Could the warning be <type> "<key>" has non-matching id "<id>" where:
   <type> is the type of item, 
   <key> is the buildObjectKey
   <id> is key
I'm using "id" here because that's in the user vocabulary. Key here is the cache key, which I understand to be the item's id.

Also, add a spectator counter around this with the item type label so it can be monitored.